pkg/apis/triggers/v1alpha1/contexts.go

```diff
@@ -0,0 +1,20 @@
+package v1alpha1
+
+import "context"
+
+// upgradeViaDefaultingKey is used as the key for associating information
+// with a context.Context. This variable doesn't really matter, so it can
+// be a total random name
+type upgradeViaDefaultingKey struct{}
+
+// WithUpgradeViaDefaulting notes on the context that we want defaulting to rewrite
+// from v1alpha1 pre-defaults to v1alpha1 post-defaults.
+func WithUpgradeViaDefaulting(ctx context.Context) context.Context {
+	return context.WithValue(ctx, upgradeViaDefaultingKey{}, struct{}{})
+}
+
+// IsUpgradeViaDefaulting checks whether we should be "defaulting" from v1alpha1 pre-defaults to
+// the v1alpha1 post-defaults subset.
```

Comment:
in these 2 comments im a bit confused by what 'defaulting" from v1alpha1 pre-defaults to the v1alpha1 post-defaults subset'  means - the quotes around "defaulting" make me also feel like im definitely missing something, maybe some more words in these docstrings could clear this up?

Reply:
Added some more comments! The two functions are to set and check whether or not the key was set. So, I added some more comments around what they key is used for i.e. upgrade default values that have changed between releases.